simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/Base64.java

```diff
@@ -0,0 +1,48 @@
+package io.prometheus.client.exporter;
+
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * This class delegates to either javax.xml.bind.DatatypeConverter (for Java < 8) or java.util.Base64 (Java 8+)
+ * to perform Base64 encoding of a String.
+ *
+ * This code requires Java 8+ for compilation.
+ */
+public class Base64 {
+
+    private static boolean hasJavaUtilBase64;
```

Comment:
Nitpick: As this is supposed to never change, it should be `final` for clarity.

Reply:
Because `hasJavaUtilBase64` is initialized in a static block, it can't be final.

Reply:
It can be if you do something like this instead:

```java
private static final boolean HAS_JAVA_UTIL_BASE64 = base64IsPresent();

private static boolean base64IsPresent() {
    try {
        Class.forName("java.util.Base64");
        return true;
    }
    catch (Throwable ex) {
        return false;
    }
}
```